docs/configuration.rst

```diff
@@ -472,3 +477,68 @@ Optional Settings
+
+.. sourcecode:: bash
+
+  docker run -d \
+    -p 127.0.0.1:8088:8088
+    -e KSQL_BOOTSTRAP_SERVERS=localhost:9092 \
+    -e KSQL_LISTENERS=http://0.0.0.0:8088 \
+    -e KSQL_KSQL_SERVICE_ID=ksql_test_service_id_ \
+    -e KSQL_KSQL_STREAMS_REPLICATION_FACTOR=3 \
+    -e KSQL_KSQL_SINK_REPLICAS=3 \
+    -e KSQL_KSQL_STREAMS_STATE_DIR=/mnt/data/ksql-server-data \
+    -v /mnt/ksql-server/data:/mnt/data/ksql-server-data/ \
+    confluentinc/cp-ksql-server:4.1.2
+
+Docker Options
+""""""""""""""
+
+* Data persistence: KSQL Server runs Kafka streams applications which may need to store persistent state. The command above maps the state store directory to a directory on the localhost, hence ensuring persistence across multiple sessions of the same container.
+* Port mapping: The above command maps the listener port to 8088 on the localhost, hence ensuring that the CLI running on the same host can connect to the ksql server.
+
+Required Settings
+"""""""""""""""""
+The following settings must be passed to run the KSQL Server image.
+
+``KSQL_BOOTSTRAP_SERVERS``
+
+  A list of host/port pairs to use for establishing the initial connection to the Kafka cluster. The client will make use of all servers irrespective of which servers are specified here for bootstrapping; this list only impacts the initial hosts used to discover the full set of servers. This list should be in the form host1:port1,host2:port2,.... Since these servers are just used for the initial connection to discover the full cluster membership (which may change dynamically), this list need not contain the full set of servers (you may want more than one, though, in case a server is down).
+
+Optional Settings
+"""""""""""""""""
+All other settings for KSQL like security, monitoring interceptors, streams application overrides can be passed to the Docker images as environment variables. The names of these environment variables are derived by replacing ``.`` with ``_``, converting the resulting string to uppercase and prefixing it with ``KSQL_``. For example, if you need to set ``ksql.streams.state.dir``, the environment variable name would be ``KSQL_KSQL_STREAMS_STATE_DIR``.
+
+The image will then convert these environment variables to corresponding KSQL Server config variables.
+
+--------
+KSQL CLI
+--------
+The KSQL CLI image will execute the ``ksql`` command and try to connect to the server at ``http://localhost:8088`` by default, as shown below. 
+
+.. sourcecode:: bash
+
+  docker run -it -rm confluentinc/cp-ksql-cli:4.1.2
+
+Note: Since the CLI is interactive, we cannot log to stdout as is customary for Docker containers. The KSQL CLI logs can instead be found at ``/var/logs/ksql-cli/``.
```

Comment:
> found at ``/var/logs/ksql-cli/``

Within the docker image, or the local host?

Reply:
@miguno you can't use `docker log` for the CLI image. This is basically impossible, since the CLI uses stdout for real output. Directing the logs to stdout as well makes it unusable.

Reply:
(docker log only emits stdout and stderr from process id 0, which is the launch command of the container).